### ISSUE TITLE
Image classification in new C++ core

### ISSUE THREAD

**Issue by wangkuiyi** [PaddlePaddle/Paddle#4531]
## Operator list
- [x] momentum_op @sidgoyal78
- [x] fc_op  @Xreki 
- [x] conv_op  @hedaoyuan @typhoonzero 
- [x] dropout @xinghai-sun 
- [x] batch_norm @jacquesqiao https://github.com/PaddlePaddle/Paddle/issues/4906
- [x] Relu @QiJune 
- [x] pool_op  @chengduoZH   
- [x] softmax  @jacquesqiao @lcy-seso @QiJune 
- [x] cross entropy @jacquesqiao @lcy-seso
- [x] accuracy @typhoonzero 

## Python train file
- [x] implement book/03.image_classification @jacquesqiao 
   - [x] VGG-16 issue https://github.com/PaddlePaddle/Paddle/issues/5100  pr https://github.com/PaddlePaddle/Paddle/pull/5192
   - [x] ResNet https://github.com/PaddlePaddle/Paddle/pull/5206
   - [x] test acc && fine turn book/03.image_classification https://github.com/PaddlePaddle/Paddle/pull/5336

## Python inference && accuracy test
- [ ] batch_norm test mode
